scipy/optimize/_qap.py

```diff
@@ -0,0 +1,722 @@
+       score: 3260
+
+    The see the relationship between the returned ``col_ind`` and ``score``,
+    use ``col_ind`` to form the best permutation matrix found, then evaluate
+    the objective function :math:`f(P) = trace(A^T P B P^T )`.
+
+    >>> n = A.shape[0]
+    >>> perm = res['col_ind']
+    >>> P = np.eye(n)[perm]
+    >>> score = int(np.trace(A.T @ P @ B @ P.T))
+    >>> print(score)
+    3260
+
+    Alternatively, to avoid constructing the permutation matrix explicitly,
+    directly permute the rows and columns of the distance matrix.
+
+    >>> score = np.trace(A.T @ B[perm][:, perm])
+    >>> print(score)
+    3260
+
+    Although not guaranteed in general, ``quadratic_assignment`` happens to
+    have found the globally optimal solution.
+
+    >>> from itertools import permutations
+    >>> perm_opt, score_opt = None, np.inf
+    >>> for perm in permutations([0, 1, 2, 3]):
+    ...     score = int(np.trace(A.T @ B[perm][:, perm]))
+    ...     if score < score_opt:
+    ...         score_opt, perm_opt = score, perm
+    >>> print(np.equal(perm_opt, res['col_ind']))
+    True
+    """
+
+    if options is None:
+        options = {}
+
+    method = method.lower()
+    methods = {"faq": _quadratic_assignment_faq,
+               "2opt": _quadratic_assignment_2opt}
+    if method not in methods:
+        raise ValueError(f"method {method} must be in {methods}.")
+    res = methods[method](A, B, **options)
+    return res
+
+
+def _common_input_validation(A, B, partial_match, maximize):
+    A = np.atleast_2d(A)
+    B = np.atleast_2d(B)
+
+    if partial_match is None:
+        partial_match = np.array([[], []]).T
+    partial_match = np.atleast_2d(partial_match)
+
+    msg = None
+    if A.shape[0] != A.shape[1]:
+        msg = "`A` must be square"
+    elif B.shape[0] != B.shape[1]:
+        msg = "`B` must be square"
+    elif A.ndim != 2 or B.ndim != 2:
+        msg = "`A` and `B` must have exactly two dimensions"
+    elif A.shape != B.shape:
+        msg = "Adjacency matrices must be of equal size"
+    elif (A < 0).any() or (B < 0).any():
+        msg = "Adjacency matrix contains negative entries"
+    elif partial_match.shape[0] > A.shape[0]:
+        msg = "There cannot be more seeds than there are nodes"
+    elif partial_match.shape[1] != 2:
+        msg = "`partial_match` must have two columns"
+    elif partial_match.ndim != 2:
+        msg = "`partial_match` must have exactly two dimensions"
+    elif (partial_match < 0).any():
+        msg = "`partial_match` contains negative entries"
+    elif (partial_match >= len(A)).any():
+        msg = "`partial_match` entries must be less than number of nodes"
+    elif not len(set(partial_match[:, 0])) == len(partial_match[:, 0]) or not \
+            len(set(partial_match[:, 1])) == len(partial_match[:, 1]):
+        msg = "`partial_match` column entries must be unique"
+
+    if msg is not None:
+        raise ValueError(msg)
+
+    if not isinstance(maximize, bool):
```

Comment:
This is too strict. `maximize=1` is quite common.

Reply:
Removed.